kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/KotlinStaticData.kt

```diff
@@ -59,7 +62,8 @@ internal class KotlinStaticData(override val generationState: NativeGenerationSt
 
         val compositeType = llvm.structType(runtime.arrayHeaderType, arrayBody.llvmType)
 
-        val global = this.createGlobal(compositeType, "")
+        val global = this.createGlobal(compositeType, "kconstarray:${arrayClass.fqNameForIrSerialization}-$numArray")
```

Comment:
LLVM will automatically add a number suffix if a global with such a name already exists.
So, the `-$numArray` part shouldn't be necessary.